Sources/Contour/Configuration/BusConfiguration.cs

```diff
@@ -555,6 +553,25 @@ public void UseRouteResolverBuilder(Func<IRouteResolverBuilder, IRouteResolver>
         public void UseSubscriptionEndpointBuilder(Func<ISubscriptionEndpointBuilder, ISubscriptionEndpoint> endpointBuilder)
         {
             this.DefaultSubscriptionEndpointBuilder = endpointBuilder;
+        }
+
+        /// <inheritdoc />
+        public void SetExcludedIncomingHeaders(IEnumerable<string> excludedHeaders)
+        {
+            var blockedHeaders = new List<string>
+                                     {
+                                         Headers.Expires,
+                                         Headers.MessageLabel,
+                                         Headers.Persist,
+                                         Headers.QueueMessageTtl,
+                                         Headers.ReplyRoute,
+                                         Headers.Timeout,
+                                         Headers.Ttl
+                                     };
+            blockedHeaders.AddRange(excludedHeaders ?? Enumerable.Empty<string>());
+            var messageHeaderStorage = new Maybe<IIncomingMessageHeaderStorage>(new MessageHeaderStorage(blockedHeaders));
+            this.SenderDefaults.IncomingMessageHeaderStorage = messageHeaderStorage;
```

Comment:
I believe the SetExcludedIncomingHeaders method is not right place to set  MessageHeaderStorage. Maybe replace this by the existing method WithIncomingMessageHeaderStorage using overload version with extra parameter 'blockingHeaders'?